src/Illuminate/Database/Schema/BlueprintState.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+namespace Illuminate\Database\Schema;
+
+use Illuminate\Database\Connection;
+use Illuminate\Database\Query\Expression;
+use Illuminate\Database\Schema\Grammars\Grammar;
+use Illuminate\Support\Fluent;
+
+class BlueprintState
+{
+    /**
+     * The blueprint instance.
+     *
+     * @var \Illuminate\Database\Schema\Blueprint
+     */
+    protected $blueprint;
+
+    /**
+     * The connection instance.
+     *
+     * @var \Illuminate\Database\Connection
+     */
+    protected $connection;
+
+    /**
+     * The grammar instance.
+     *
+     * @var \Illuminate\Database\Schema\Grammars\Grammar
+     */
+    protected $grammar;
+
+    /**
+     * The columns.
+     *
+     * @var \Illuminate\Database\Schema\ColumnDefinition[]
+     */
+    private $columns = [];
+
+    /**
+     * The primary key.
+     *
+     * @var null|\Illuminate\Database\Schema\IndexDefinition
+     */
+    private $primaryKey;
+
+    /**
+     * The indexes.
+     *
+     * @var \Illuminate\Database\Schema\IndexDefinition[]
+     */
+    private $indexes = [];
+
+    /**
+     * The foreign keys.
+     *
+     * @var \Illuminate\Database\Schema\ForeignKeyDefinition[]
+     */
+    private $foreignKeys = [];
+
+    /**
+     * Create a new blueprint state instance.
+     *
+     * @param  \Illuminate\Database\Schema\Blueprint  $blueprint
+     * @param  \Illuminate\Database\Connection  $connection
+     * @param  \Illuminate\Database\Schema\Grammars\Grammar  $grammar
+     * @return void
+     */
+    public function __construct(Blueprint $blueprint, Connection $connection, Grammar $grammar)
+    {
+        $this->blueprint = $blueprint;
+        $this->connection = $connection;
+        $this->grammar = $grammar;
+
+        $schema = $connection->getSchemaBuilder();
+        $table = $blueprint->getTable();
+
+        $this->columns = collect($schema->getColumns($table))->map(fn ($column) => new ColumnDefinition([
+            'name' => $column['name'],
+            'type' => $column['type_name'],
+            'full_type_definition' => $column['type'],
+            'nullable' => $column['nullable'],
+            'default' => is_null($column['default']) ? null : new Expression($column['default']),
+            'autoIncrement' => $column['auto_increment'],
+            'collation' => $column['collation'],
+            'comment' => $column['comment'],
+            'virtualAs' => ! is_null($column['generation']) && $column['generation']['type'] === 'virtual'
```

Comment:
Shorter form:
`
  'virtualAs' =>($column['generation']['type'] ?? '') === 'virtual'
                ? $column['generation']['expression'] : null,
 'storedAs' => ($column['generation']['type'] ?? '') === 'stored'
                ? $column['generation']['expression'] : null,
`

Reply:
It's just moved from here: https://github.com/laravel/framework/pull/51373/files#diff-7cf2f940b724463282b1dbae24464cb3a7c7c0864d40b4c3706eb99f65994d2cL279